# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/dns/nsHostResolver.h
# Commit: 6f4ca967aa44
# Full Hash: 6f4ca967aa446cf24f4224576f51803c982f3d7b
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2024-06-14 20:49:45
# Regressor Bug: 1901990
# File Overlap Count: 2
# Description:
#   Bug 1901990 - Introduce a pref for setting a mock HTTPS RR, r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D213559
# ==============================================================================

diff -r 8a2bdb44382a -r 6f4ca967aa44 netwerk/dns/nsHostResolver.h
--- a/netwerk/dns/nsHostResolver.h	Fri Jun 14 13:27:36 2024 +0000
+++ b/netwerk/dns/nsHostResolver.h	Fri Jun 14 13:42:37 2024 +0000
@@ -146,6 +146,11 @@
                                                     nsresult* aRv);
 
   /**
+   * return a mock HTTPS record
+   */
+  already_AddRefed<nsHostRecord> InitMockHTTPSRecord(const nsHostKey& key);
+
+  /**
    * removes the specified callback from the nsHostRecord for the given
    * hostname, originAttributes, flags, and address family.  these parameters
    * should correspond to the parameters passed to ResolveHost.  this function